docs/how-to/uc-image-creation.rst

```diff
@@ -0,0 +1,307 @@
+How to create a real-time Ubuntu Core image
+===========================================
+
+.. note::
+
+    This guide assumes access to features available exclusively to `dedicated Snap Store`_ users.
+
+Canonical builds and publishes Ubuntu Core images for a range of `supported platforms`_.
+However, these images include the generic Linux kernel. 
+In order to run Ubuntu Core with the real-time kernel, we need to build it ourselves. 
+
+This guide shows how to build an Ubuntu Core image with the real-time kernel.
+We start by showing how to :ref:`build a vanilla image <ubuntu-core-image>` which is excellent for testing and tuning, and then illustrate how to :ref:`make a custom image <custom-ubuntu-core-image>` with production-ready kernel configurations.
+
+
+.. _ubuntu-core-image:
+
+Create an image with the real-time kernel
+-----------------------------------------
+
+To do this we need to describe the image content and then use a tool to build it.
+
+.. _model-assertion:
+
+Create the model assertion
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Inside an empty directory, create a file named ``model.json`` with the above content.
+
+Change the following:
+
+- ``authority-id``, ``brand-id`` to your developer ID, since this is custom model. Use ``snapcraft whoami`` command to get your developer ID.
+- ``timestamp`` to an RFC3339 formatted string that falls within the validity period of your signing key. Use ``date -Iseconds --utc`` command to generate the current time.
+- ``store`` to your dedicated Snap Store ID.
+- ``model`` to a name that accurately represents your device(s).
+
+The ``snaps`` array is a list of snaps that get included in the image.
```

Comment:
Hi @farshidtz , I would suggest rephrase this slightly to bring focus to the real-time kernel and let the upstream UC docs do the rest of the explanation as needed.

```
The ``realtime-kernel`` snap must be defined in the `snaps array`_ for the
real-time kernel to be included in the image. You can add other required snaps
(e.g. snaps for your real-time applications) to this list.

.. _snaps array: https://ubuntu.com/core/docs/create-model-assertion#heading--edit-model-3
```

But please feel free to leave as is if you disagree.
Thanks.

Reply:
I'd leave this as it is for the following reasons:
- The reader isn't expected to change anything here as the yaml already has the realtime-kernel snap listed.
- The realtime-kernel snap may contain more than just the kernel such as hardcoded RT configurations.